packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1783,6 +1783,21 @@ export class TransactionController extends BaseControllerV1<
     const initialTx = listOfTxParams[0];
     const common = this.getCommonConfiguration(initialTx.chainId);
 
+    // We need to ensure we get the nonce using the the NonceTracker on the chain matching
+    // the txParams. In this context we only have chainId available to us, but the
+    // NonceTrackers are keyed by networkClientId. To workaround this, we attempt to find
+    // a networkClientId that matches the chainId. As a fallback, the globally selected
+    // network's NonceTracker will be used instead.
+    let networkClientId: NetworkClientId | undefined;
+    try {
+      networkClientId = this.messagingSystem.call(
+        `NetworkController:findNetworkClientIdByChainId`,
+        initialTx.chainId,
+      );
```

Comment:
Can we add this to `AllowedActions` by importing from `NetworkControllerFindNetworkClientIdByChainIdAction` so that we know/track the actions controller take?

Reply:
Oh I noticed this PR is going to merge into multichain branch, please don't mind my comment in this case, rest will be reviewed anyway.